### PR TITLE
fix: Force partition key parameter and Fix hierarchical partition key deployment and analytical store enablement 

### LINKED ISSUE #1222: [AVM Module Issue]: PartitionKey should be required by the container resource for a NoSQL database
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

This was reported in the CARML repository and hasn't been solved so far. This module inherited this bug as part of the porting
https://github.com/Azure/ResourceModules/issues/4278

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1225: [AVM Module Issue]: Cannot enable analytical store since there is no way to enable synapse link which is a pre-requisite
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

To enable analytical store in a container you need to specify the TTL you would like to have. Also, you need to enable Synapse analytics links in the cosmosdb account. If you don't do this a request to create a container with analytical store enable will fail if it sees a value in the analyticalTTL property

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1226: [AVM Module Issue]: Cannot only specify throughput at container level or database level
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

Right now, database and container throughput default value is 400 RU in the current version. Making impossible to only specify throughput at container or database since even if you specify that you want a database of 1000 RU the container will still default to 400 thus generating unnecesary costs

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1227: [AVM Module Issue]: Support IntelliSense when creating account, sql database and containers
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

Right now the module is not using user defined types, which makes the experience of using the module poor, as you won't know what the required and optional properties and their default values are.

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1228: [AVM Module Issue]: Allow disabling local auth
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

Allow to disable local key authentication for those who only want to use managed identities.

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1229: [AVM Module Issue]: Cannot enable hierarchical partitions keys as it needs partition key version 2
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

Even though the module allows to enable hierarchical partition keys, there is not a possible way to do it as it requires you to specify version 2 of the partition key and the user doesn't have any way to do it

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #4278: [Bug Report]: paritionKey should be required by the container resource for a NoSQL database
### Describe the bug

The [NoSQL Container resource](https://github.com/Azure/ResourceModules/tree/main/modules/document-db/database-account/sql-database/container) requires properties.resources.partitionKey but it isn't marked as required. The API rejects a container with a partition key with this error:
> A Partition key definition is not specified in the request.

### To reproduce

Deploy a container to a NoSQL database without specifying the partitionKey.

### Code snippet

```Bicep
param databaseAccountName string
param databaseName string
param containerName string

resource cosmosDbAccount 'Microsoft.DocumentDB/databaseAccounts@2023-09-15' existing = {
  name: databaseAccountName
}

resource alertDatabase 'Microsoft.DocumentDB/databaseAccounts/sqlDatabases@2023-09-15' = {
  name: databaseName
  parent: cosmosDbAccount
  properties: {
    resource: {
      id: databaseName
    }
  }
}

resource alertSpecificationContainer 'Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers@2023-09-15' = {
  name: containerName
  parent: alertDatabase
  properties: {
    resource: {
      id: containerName
    }
  }
}
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ghost**
## Description

- Added more tests covering most NoSQL API and Account focused configurations. 
- Removed non-related NoSQL API tests from its test suite (They were already covered by waf-aligned and plain test suite).
- Make partition key path parameter mandatory. It couldnt be optional in the first place anyways as it gives an error in deployment time
     - Fixes #1222 
- Add "/" to partition key path if the user doesn't specify it to avoid errors when it's missing
- Added user defined types for better consumer experience and intelisense. 
     - Closes #1227 
- Fixed an issue where deployment of a hierarchical partition key would fail even though it was allowed through parameter. 
     - Fixes #1229 
- Fixed an issue where you were forced to have RUs both in Container and Database due to default values being used. It was impossible to have RUs only at one level 
     - Fixes #1226 
- Fixed an issue where analytical store ttl couldn't be enabled since there was missing a parameter (enableAnalyticalStore) at account level 
     - Fixes #1225 
- Allow disabling local auth for NoSQL API as it is the recommended configuration when using managed identities
     - Closes #1228 

Closes https://github.com/Azure/ResourceModules/issues/4278

## Remarks
- I have had to hardcode the region where the tests are being deployed to eastus as the pipeline is picking up a random region which might have new cosmosdb constraints or not supported features that the tests need. In my opinion is better to hardcode the region rather than remove the tests. I didn't find a way to tell the pipeline to deploy to specific regions without affecting all pipelines.

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.document-db.database-account](https://github.com/bryansan-msft/fork-bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml)     |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [x] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/document-db/database-account/main.bicep [hunk 1]
```diff
@@ -30,26 +30,32 @@ param locations array
   'BoundedStaleness'
   'Strong'
 ])
-@description('Optional. The default consistency level of the Cosmos DB account.')
+@description('Optional. Default to Session. The default consistency level of the Cosmos DB account.')
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/main.bicep:33]:
```suggestion
@description('Optional. The default consistency level of the Cosmos DB account.')
```

The default is automatically documented. Adding it also to the description means to start maintaining 2 locations instead of one. Please also update for the other parameters.

File: avm/res/document-db/database-account/tests/e2e/gremlindb/main.test.bicep [hunk 2]
```diff
@@ -20,6 +20,9 @@ param serviceShort string = 'dddagrm'
 @description('Optional. A token to inject into the name of each resource.')
 param namePrefix string = '#_namePrefix_#'
 
+// Pipeline is selecting random regions which dont support all cosmos features and have constraints when creating new cosmos
+var eastUsResourceLocation = 'eastus'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/gremlindb/main.test.bicep:24]: In alignment with purview, I'd suggest to call it `enforcedLocation` if you'd agree (across all test files).

File: avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep [hunk 3]
```diff
@@ -124,3 +122,181 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
     diagnosticDependencies
   ]
 }]
+
+module enableSystemAssignedManagedIdentity '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep:126]: Please note - if you add multiple test deployments to this file, none of them will show up in the readme. Personally, I'd suggest to have a test file / folder each. I'd also be curious why not having the parameters you test all being part of the same test - unless they're not compatible. If the latter is the case, you could try to split seperate test cases off where required.

Comment (AlexanderSehr) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep:126]: @eriqua what are your thoughts on the changes? I guss ultimately it's up to the owner, but it would be nice to align with the convention, I guess.

File: avm/res/document-db/database-account/main.bicep (hunk 1 above)
Comment (bryansan-msft) [reply avm/res/document-db/database-account/main.bicep:33]: Where is it documented? In the auto generated read me file?
Dont you think it is a better consumer experience to see it while you are writing code rather than having to jump to the documentation? In the end the readme is generated automatically and we are forcing people to regenerate it. So I would rather say that we have just one place to maintain.

In my opinion. I'm more on the side of forcing the developer to jump to the documentation as minimum as possible. They might not even know about it. Look at me. This is my first time consuming this module and I didn't even know the defaults were there and the first place i looked up was in the intelisense.

If you still consider not to have it in the description then I will make the change but I unfortunately dont agree with it

File: avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep (hunk 3 above)
Comment (bryansan-msft) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep:126]: In my opinion, I don't mind if they don't appear in the Readme file. In fact, I am happy if they don't appear in it as I created these to have a more complete test suite rather than having more examples for the user. In the end, the lack of these tests is what provoked these bugs. 

Some of those settings are incompatible with each other so I can't test them all in only one account. Yes, I could create a test folder for each of them but think that I would then have a LOT of test folders since I have many tests for multiple scenarios and there are still many more which I didn't create just because the PR was already big.

Comment (bryansan-msft) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep:126]: Another thing worth to mention is that is always easier to find bugs or tests features with tiny tests rather than big bang ones. Since it was very but very hard to find out what was happening with the analytical store + the time taken to deploy everything in one go with the pre PR sql test suite

File: avm/res/document-db/database-account/main.bicep (hunk 1 above)
Comment (bryansan-msft) [reply avm/res/document-db/database-account/main.bicep:33]: Second thoughts about this. I have one use case where I don't have the real default value used in the parameter so it is really not documented in the Readme.md file as it is not picked up by the generation. It is not specified in the parameters as it created the bug #1226 the default value needs to be calculated on the fly

File: avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep (hunk 3 above)
Comment (bryansan-msft) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep:126]: I will create all the tests folder then

Comment (AlexanderSehr) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep:126]: If you need a hand, just say the word

File: avm/res/document-db/database-account/tests/e2e/consistency/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module boundedStalenessConsistency '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/consistency/main.test.bicep:38]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/consistency/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,60 @@
+targetScope = 'subscription'
+
+metadata name = 'bounded'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/consistency/main.test.bicep:3]:
```suggestion
metadata name = 'Using bounded consistency'
```

File: avm/res/document-db/database-account/tests/e2e/continousBckup/main.test.bicep [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module enableContinousBackup '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/continousBckup/main.test.bicep:38]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/continousBckup/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,59 @@
+targetScope = 'subscription'
+
+metadata name = 'continousBckup'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/continousBckup/main.test.bicep:3]:
```suggestion
metadata name = 'Enabling continuous backups'
```

File: avm/res/document-db/database-account/tests/e2e/disableLocal/main.test.bicep [hunk 8]
```diff
@@ -0,0 +1,57 @@
+targetScope = 'subscription'
+
+metadata name = 'disable Local'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/disableLocal/main.test.bicep:3]:
```suggestion
metadata name = 'Using Entra ID authentication'
```

File: avm/res/document-db/database-account/tests/e2e/disableLocal/main.test.bicep [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module disableLocalAuth '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/disableLocal/main.test.bicep:37]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/autoFailoverOff/main.test.bicep [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module disableAutomaticFailover '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/autoFailoverOff/main.test.bicep:52]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/analytical/main.test.bicep [hunk 11]
```diff
@@ -0,0 +1,58 @@
+targetScope = 'subscription'
+
+metadata name = 'analytical'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/analytical/main.test.bicep:3]:
```suggestion
metadata name = 'Using analytical storage'
```

File: avm/res/document-db/database-account/tests/e2e/periodicBckup/main.test.bicep [hunk 12]
```diff
@@ -0,0 +1,61 @@
+targetScope = 'subscription'
+
+metadata name = 'periodicBckup'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/periodicBckup/main.test.bicep:3]:
```suggestion
metadata name = 'Using periodic backups'
```

File: avm/res/document-db/database-account/tests/e2e/periodicBckup/main.test.bicep [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module enablePeriodicBackup '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/periodicBckup/main.test.bicep:38]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/systemMi/main.test.bicep [hunk 14]
```diff
@@ -0,0 +1,55 @@
+targetScope = 'subscription'
+
+metadata name = 'systemMi'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/systemMi/main.test.bicep:3]:
```suggestion
metadata name = 'Deploying with a System-Assigned identity'
```

File: avm/res/document-db/database-account/tests/e2e/systemMi/main.test.bicep [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module enableSystemAssignedManagedIdentity '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/systemMi/main.test.bicep:38]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/userMi/main.test.bicep [hunk 16]
```diff
@@ -0,0 +1,73 @@
+targetScope = 'subscription'
+
+metadata name = 'userMi'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/userMi/main.test.bicep:3]:
```suggestion
metadata name = 'Deploying with a user-assigned identity'
```

File: avm/res/document-db/database-account/tests/e2e/userMi/main.test.bicep [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module enableUserAssignedManagedIdentity '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/userMi/main.test.bicep:51]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/zoneRedundant/main.test.bicep [hunk 18]
```diff
@@ -0,0 +1,52 @@
+targetScope = 'subscription'
+
+metadata name = 'zoneRedundant'
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/zoneRedundant/main.test.bicep:3]:
```suggestion
metadata name = 'Using zone redundancy'
```

File: avm/res/document-db/database-account/tests/e2e/zoneRedundant/main.test.bicep [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module enableZoneRedundant '../../../main.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/document-db/database-account/tests/e2e/zoneRedundant/main.test.bicep:38]:
```suggestion
module testDeployment '../../../main.bicep' = {
```
That's why the script didn't work. As per the specs (a bit hidden maybe) it must be the above name to identify the relevant test 'block'

File: avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep (hunk 3 above)
Comment (AlexanderSehr) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep]: Say no more :)

I added suggestions that would take care of the issue. Once applied, you should be able to run the script without an issue :)

For transparency, this would be the corresponding spec: https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpnfr12---category-testing---deployment-test-naming

File: avm/res/document-db/database-account/main.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/document-db/database-account/main.bicep:33]: Hey @bryansan-msft,
I don't think it's documented anywhere as it is no spec - I guess it's more of a recommendation to keep the maintenance effort at bay.

To your question you can find the metadata actually in a few locations. One is the readme, yes - but as you said, you don't want to jump back and forth. That said, the Bicep Linter got you covered too like for example this:

image

with the 'allowed values'.

Contrary to my expectations, you're right with the defaults. I don't know why - but they are not shown. Like here where it should say `true`:
image

I agree with your observation, but, would actually want to see this in linter more than seeing it in the description. I'll check if I can find an issue for that. Afterall - all the other stuff is shown, why not the default?

cc: @eriqua, @ChrisSidebotham

Comment (AlexanderSehr) [reply avm/res/document-db/database-account/main.bicep:33]: Update: I wasn't able to find an issue (suprisingly) and I'll open one

Comment (eriqua) [reply avm/res/document-db/database-account/main.bicep:33]: Descriptions for input parameters are not that strictly limited currently. For example, we have allowed values ported there in some cases, while that is already available as Alex brought up above.
That said, I agree with @bryansan-msft on it would be better to show default values via intellisense and I agree with @AlexanderSehr on this is more a feature request for linter than it is for the avm description metadata as this may be challenging to maintain overtime and also to require consistently for all applicable input parameter descriptions. 
I'd suggest to first understand how long it would take realistically to implement that feature. Based on that we may want to provide guidelines to contributors on not to waste effort on explicitly call out allowed or default values in description metadata.

File: avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep (hunk 3 above)
Comment (eriqua) [reply avm/res/document-db/database-account/tests/e2e/plain/main.test.bicep]: > @eriqua what are your thoughts on the changes? I guss ultimately it's up to the owner, but it would be nice to align with the convention, I guess.

I see the discussion progressed to an agreement for dedicated test folders already. Agree with the agreement 🤝 😆

File: avm/res/document-db/database-account/main.bicep (hunk 1 above)
Comment (bryansan-msft) [reply avm/res/document-db/database-account/main.bicep:33]: Well, I agree that if the linter does it it would be better for everyone. But until it does it what do we do? I am keen on leaving them on the description but I am not the one deciding.

Comment (bryansan-msft) [reply avm/res/document-db/database-account/main.bicep:33]: One thing to add is that if someone does this then the default value would be missing completely from everywhere as the auto generated Readme will not pick it up

image

**Comment by bryansan-msft** [comment]
@AlexanderSehr Readme problem is already solved, changes are already implemented and pipeline is still green.

File: avm/res/document-db/database-account/main.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/document-db/database-account/main.bicep:33]: correct @bryansan-msft . In somecases it's implemented like this because UDTs don't support defaults (e.g. in the diagnostic settings), in others because other logic is required. It's unfortunate, but I wouldn't know how to solve this in a good way.

Coming back to the original issue - I won't block adding the defaults to the description. Ultimately, it's your choice and if you feel it's worth-while, so be it :)

Resolving the comment.

**Review by AlexanderSehr — APPROVED** [review]
